services/gitdiff/gitdiff.go

```diff
@@ -1136,6 +1134,14 @@ func GetDiff(ctx context.Context, gitRepo *git.Repository, opts *DiffOptions, fi
 			AddArguments(opts.WhitespaceBehavior...).
 			AddDynamicArguments(actualBeforeCommitID, opts.AfterCommitID)
 		opts.BeforeCommitID = actualBeforeCommitID
+
+		if opts.BeforeCommit == nil {
+			var err error
+			opts.BeforeCommit, err = gitRepo.GetCommit(opts.BeforeCommitID)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Why I insist to revert the `DiffOptions` fields, here is a real example:

The `BeforeCommitID` is updated to a new `actualBeforeCommitID`. But if caller already passed `BeforeCommit`, then this `if` is skipped, then `BeforeCommit` and `BeforeCommitID` mismatch.

The new fields would cause new regression bugs like this, because it is really difficult to keep their values sync.